examples/assembly/addarrays.s

```diff
@@ -0,0 +1,116 @@
+.globl addarrays_serial
+.type addarrays_serial, %function
+
+# void addarrays_serial(int *res, int *A, int *B, int N)
+# Set res[i] = A[i] + B[i] for i from [0, N)
+# x0 -> res
+# x1 -> A
+# x2 -> B
+# w3 -> N
+# w4 -> loop counter
+
+addarrays_serial:
+    mov w4, wzr
+.loop_begin:
+    cmp w4, w3
+    beq .loop_exit
+    ldr w5, [x1, w4, uxtw 2]
+    ldr w6, [x2, w4, uxtw 2]
+    add w5, w5, w6
+    str w5, [x0, w4, uxtw 2]
+    add w4, w4, 1
+    b .loop_begin
+.loop_exit:
+    ret
+
+.global addarrays_sve
+.type addarrays_sve, %function
+
+# void addarrays_sve (int *restrict res, int *A, int *B, int N)
+# Set res[i] = A[i] + B[i] for i from [0, N) using SVE vectors.
```

Comment:
I would add to this something to explain that we're using them in a scalable way. Which seems redundant but you can have fixed size SVE code, and it would make the comments in the code below more understandable.